src/formatDate.js

```diff
@@ -8,7 +8,46 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+
+  const dateInOldFormatArr = date.split(oldSeparator);
+  const dateInNewFormatArr = [];
+
+  const month = dateInOldFormatArr[fromFormat.indexOf('MM')];
+  const day = dateInOldFormatArr[fromFormat.indexOf('DD')];
+  let year;
+
+  if (date.length === 10) {
+    year = dateInOldFormatArr[fromFormat.indexOf('YYYY')];
+    year = `${year[2]}${year[3]}`;
+  } else if (date.length === 8) {
+    year = dateInOldFormatArr[fromFormat.indexOf('YY')];
+  }
+
+  for (const ch of toFormat) {
```

Comment:
The for-loop iterates over each character in the `toFormat` array, but it should iterate over each format element (e.g., 'YYYY', 'MM', 'DD'). You should be checking the entire format string, not just if it includes a 'M', 'D', or 'Y'.